manifests/init.pp

```diff
@@ -16,7 +16,7 @@
   String                    $certificate_checker_path,
 
   String $logfile = '/var/log/certificate-checker.jsonl',
-  String $ensure = 'installed',
+  String $ensure = '1.2.0',
```

Comment:
Should it be there? How its related to commit log?

Reply:
`--ignore-nonexistent` was [added in certificate-checker-1.2.0](https://github.com/smortex/certificate-checker/blob/master/CHANGELOG.md#120), one can still override this by passing `$certificate_checker::ensure` an explicit value, but on a configured node, only `latest` and `1.2.0` will work (previous install would keep the previous version and fail to start with the new args).

That's far from ideal, but I could not think of a better way to handle this. Any idea?

Reply:
Using a fact to know installed version and fail is requirements are not met.
This approach should reduce the need to bump version here each release of the tool (ie. certificate-checker).

Reply:
That would require to know where to look for certificate-checker, which facter does not know about.

I consider reverting this to the previous `installed` value, and making `--ignore-nonexistent` optional with a new parameter defaulting to false. This may be switched to true in a future major release of the module.

Looks good?

Reply:
Just did it, resolving the discussion.

Reply:
If `certificate-checker' is in path, you could use ":confine" directive in the embedded ruby fact.